packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,21 +1,246 @@
+}
+
+const copySuccessTooltip = (
+  <>
+    {successIcon}
+    {`Link copied`}
+  </>
+)
+
+export function LinkIndicatorButton(props) {
+  const { orgId, siteId, buildId, active } = props
+  const [linkButtonCopyProps, setLinkButtonCopyProps] = useState()
+
+  const copyLinkClick = () => {
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_CLICK`,
+      orgId,
+      siteId,
+      buildId,
+      name: `copy link`,
+    })
+
+    setLinkButtonCopyProps({
+      tooltipContent: copySuccessTooltip,
+      overrideShowTooltip: true,
+    })
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({
+        tooltipContent: copySuccessTooltip,
+        overrideShowTooltip: false,
+      })
+      // We want the tooltip to linger for two seconds to let the user know it has been copied
+    }, 2000)
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({ tooltipContent: `Copy Link` })
+      // The tooltips fade out, in order to make sure that the text does not change
+      // while it is fading out we need to wait a bit longer than the time used above.
+    }, 2400)
+
+    if (window) {
+      navigator.clipboard.writeText(window.location.href)
+    }
+  }
+
+  const trackHover = () => {
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `link hover`,
+    })
+  }
+
+  return (
+    <IndicatorButton
+      testId={`link`}
+      iconSvg={linkIcon}
+      onClick={copyLinkClick}
+      onMouseEnter={trackHover}
+      {...props}
+      {...linkButtonCopyProps}
+    />
+  )
+}
+
+export function InfoIndicatorButton(props) {
+  const { active, orgId, siteId, buildId } = props
+  const trackHover = () => {
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `info hover`,
+    })
+  }
+  return (
+    <IndicatorButton
+      testId="info"
+      iconSvg={infoIcon}
+      onMouseEnter={trackHover}
+      {...props}
+    />
+  )
+}
+
+export function BuildErrorIndicatorTooltip({ siteId, orgId, buildId }) {
+  return (
+    <>
+      {failedIcon}
+      {`Unable to build preview`}
+      <a
+        href={generateBuildLogUrl({ orgId, siteId, buildId })}
+        target="_blank"
+        rel="noreferrer"
+        onClick={() => {
+          trackEvent({
+            eventType: `PREVIEW_INDICATOR_CLICK`,
+            orgId,
+            siteId,
+            buildId,
+            name: `error logs`,
+          })
+        }}
+        data-gatsby-preview-indicator="tooltip-link"
+      >
+        <p data-gatsby-preview-indicator="tooltip-link-text">{`View logs`}</p>
+        <div data-gatsby-preview-indicator="tooltip-svg">{logsIcon}</div>
+      </a>
+    </>
+  )
+}
+
+function BuildSuccessIndicatorTooltip({
+  isOnPrettyUrl,
+  sitePrefix,
+  orgId,
+  siteId,
+  buildId,
+}) {
+  return (
+    <>
+      {`New preview available`}
+      <a
```

Comment:
A tags need an href in the DOM spec. use a button instead or add a href="#"

Reply:
Good call